libs/ui-toolkit/datepicker/src/lib/calendar-header.component.ts

```diff
@@ -28,12 +32,43 @@ export class CalendarHeaderComponent<D> extends MatCalendarHeader<D> {
    */
   @Input() public colorCtrls = '#888E8E';
 
-   constructor(
+  /**
+   * A flag to choose the behavior of datepicker header
+   * If it's set year-only, then the header can't be changed
+   * If it's set year-month, then the hader can toggle between year and multi-year view
+   * If it's set all, then the default behavior will be chosen
+   *
+   * @default "all"
+   */
+  @Input() public mode: DatePickerMode = 'all';
+
+  constructor(
     _intl: MatDatepickerIntl,
     @Inject(forwardRef(() => MatCalendar)) calendar: MatCalendar<D>,
     @Optional() _dateAdapter: DateAdapter<D>,
     @Optional() @Inject(MAT_DATE_FORMATS) _dateFormats: MatDateFormats,
-    changeDetectorRef: ChangeDetectorRef) {
-      super(_intl, calendar, _dateAdapter, _dateFormats, changeDetectorRef);
+    changeDetectorRef: ChangeDetectorRef,
+    private modeService: DatepickerModeService
+  ) {
+    super(_intl, calendar, _dateAdapter, _dateFormats, changeDetectorRef);
   }
+
+  public ngOnInit() {
+    this.modeService.listen().pipe(untilDestroyed(this)).subscribe((mode: DatePickerMode) => this.mode = mode);
+  }
+
+  public currentPeriodClicked = (): void => {
+    if (this.mode === 'year-only') {
+      this.calendar.currentView = 'multi-year';
+      return;
+    }
+
+    if (this.mode === 'year-month') {
+      this.calendar.currentView = this.calendar.currentView === 'multi-year' ? 'year' : 'multi-year';
+      return;
+    }
```

Comment:
You could also use a `switch` instead of checking `mode` twice

Reply:
ok